altair/vegalite/v5/api.py

```diff
@@ -1998,13 +2030,13 @@ def transform_filter(
             returns chart to allow for chaining
         """
         if isinstance(filter, Parameter):
-            new_filter: dict[str, bool | str] = {"param": filter.name}
+            new_filter: dict[str, Any] = {"param": filter.name}
             if "empty" in kwargs:
                 new_filter["empty"] = kwargs.pop("empty")
             elif isinstance(filter.empty, bool):
                 new_filter["empty"] = filter.empty
-            filter = new_filter  # type: ignore[assignment]
```

Comment:
https://github.com/vega/altair/pull/3480/files/419a4e944f231026322e2c4f137e7a3eb94735e8#r1679700744 

@binste 
> Why does this have to be an Any type? Can FilterTransform actually accept anything other than `bool` or `str`?

That one was a dual fix for the type ignores above and below this comment

Reply:
Not sure I can fully follow but if it gets rid of some type ignores I trust that it makes sense :)